sci-wcalc/coplanar_syn.man

.\" Copyright (c), 2005 Dan McMahill
.\" Do not edit this directly.  Edit the XML source file instead
.\"

.TH coplanar_syn "" "" "Wcalc" "Wcalc Commands"
.B coplanar_syn
- Single coplanar transmission line synthesis

.SH SYNOPSIS

[w_out, h_out, l_out, er_out] = 
coplanar_syn(z0, w, h, l, tmet, rho, rough, er, tand, f, flag)


.SH INPUT PARAMETERS

.TP 10
z0
: desired characteristic impedance (Ohms)
.TP 10
w
: width of coplanar line (meters)
.TP 10
h
: substrate thickness (meters)
.TP 10
l
: lentgh of coplanar line (meters)
.TP 10
tmet
: metal thickness (meters)
.TP 10
rho
: resistivity of the metalization (relative to copper)
.TP 10
rough
: conductor surface roughness (meters-rms)
.TP 10
er
: substrate relative permitivitty
.TP 10
tand
: substrate loss tangent
.TP 10
f
: synthesis frequency (Hz)
.TP 10
flag
: indicates which parameters should be optimized
    
.RS
.TP 10
flag=1
: synthesize the metal width
.TP 10
flag=1
: synthesize the substrate thickness
.TP 10
flag=2
: synthesize the substrate dielectric constant
.TP 10
flag=3
: synthesize the metal length
.RE


.SH OUTPUT VALUES

.TP 10
w_out
: Calculated width of coplanar line (meters)
.TP 10
h_out
: Calculated height of coplanar line (meters)
.TP 10
l_out
: Calculated length of coplanar line (meters)
.TP 10
er_out
: Calculated substrate relative permitivitty of coplanar line
.SH DESCRIPTION

Function to synthesize the physical dimenstions of a single
coplanar transmission line to achieve a desired characteristic
impedance and electrical length.  The desired characteristic
impedance and some of the physical
dimensions of the coplanar line are given as inputs and the
remaining dimensions are calculated.


.NF



                 |<--W-->|
                  _______    
                 | metal | <- rho,rough
    ----------------------------------------------
   (  dielectric                         /|\     (
    )   er,tand                       H   |       )
   (                                     \|/     (
    ----------------------------------------------
    XXXXXXXXXXXXXXXXXX ground XXXXXXXXXXXXXXXXXXXX




.FI

.SH EXAMPLE
.nf

.TP
 // mils to meters conversion factor
sf=25.4e-6;
.TP
 // desired impedance
z0=75;
.TP
 // desired electrical length (degrees)
elen=90;
.TP
 // width (we'll synthesize this)
w=0;
.TP
 // height
h=62*sf;
.TP
 // length
l=1100*sf;
.TP
 // metal thickness
tmet=1.4*sf;
.TP
 // resitivity
rho=1.72e-8;
.TP
 // metal surface roughness
rough=0.05*sf;
.TP
 // relative dielectric constant
er=4.5;
.TP
 // loss tangent
tand=0.01;
.TP
 // frequency
f=2.4e9;
.TP
 // synthesize width
flag=0;
[w_out,h_out,l_out,er_out] = ...
  coplanar_syn(z0,elen,w,h,l,tmet,rho,rough,er,tand,f,flag);
disp(sprintf('w=%g mils, l=%g mils',w_out/sf,l_out/sf));
disp(sprintf('er=%g',er_out));
disp(sprintf('z0=%g ohms, length=%g degrees',z0,elen));
.fi
.SH SEE ALSO
stdio-wcalc(1),
air_coil_calc(), air_coil_syn(), coax_calc(), coax_syn(), coplanar_calc(), coplanar_syn(), coupled_microstrip_calc(), coupled_microstrip_syn(), coupled_stripline_calc(), coupled_stripline_syn(), ic_microstrip_calc(), ic_microstrip_syn(), microstrip_calc(), microstrip_syn(), stripline_calc(), stripline_syn()
.SH AUTHOR

Dan McMahill

.SH BUGS

None known


NEWS
$Id: NEWS,v 1.10 2009/02/28 03:26:18 dan Exp $

Wcalc -- history of user-visible changes. 

Copyright (c) 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005
2006, 2007, 2008, 2009 Dan McMahill

Please report bugs on the sourceforge project page.

===============
Version 1.1
===============

Added series/parallel RC equivalent circuit calculator.

Added series/parallel RL equivalent circuit calculator.

Added self and mutual inductance of two rectangular bars.

Added coplanar waveguide model.

Added coupled stripline model.

Corrected the Q calculations for low frequency (not skindepth) region
in air core inductors.  Also corrected Q calculation for conductors
other than copper.

Corrected the calculation of incremental conductance for the
microstrip model.

Corrected a bug in the calculation of conductor losses in the
microstrip model.

Added/enabled code for calculating conductor losses in the coupled
microstrip model.

When compiling with gtk-2.10 or newer, use gtkPrint for printing.
This gives a more professional/standard print dialog and also enables
printing under windows.

Added a preliminary Dutch translation.

Converted the figures to all use solid grayscale fills instead of
pattern fills as the former prints better.

Several updates to the build system to moderize it a bit.  Uses
AM_CONDITIONALS more appropriately, uses AC_PROG_LIBTOOL instead of
AM_PROG_LIBTOOL, etc.

Update the win32 build script to generate an installer that works
correctly under vista.

Add .wc file associations under vista.

Update the octave build infrastructure to work with octave-3.0 and
newer where mex support is built into mkoctfile instead of a
standalone mex shell script.

Switch to GPL licensing.

Create several utility functions to make it faster to build new gtk
forms.  This is only visible to developers.

Improved autogen.sh to check versions of several of the auto* tools to
help make it easier to diagnose any issues which may come up.

Repair compilation with the SunPRO compilers.


===============
Version 1.0
===============
Replaced the alert dialog box in the gtk version with a log
window.  This prevents a problem when many alerts are generated
and the user is presented with many many alert windows.

Added win32 support along with a build script for a windows installer.

Fixed a bug which gave lots of error/warning messages in the cgi
frontend for coupled microstrip.

===============
Version 0.9
===============
First public release.

===============
Version 0.1 
===============
Initial tagged version.  This does good analysis and synthesis
of microstrip.  Considered stable for this purpose.  However,
open/save/print and units other than mils are not supported yet.



octave-wcalc/coplanar_syn.m
%COPLANAR_SYN     - Single coplanar transmission line synthesis 
%  
%  SYNOPSIS
%  
%  [w_out, h_out, l_out, er_out] = coplanar_syn(z0, w, h, l, tmet, rho, rough, er,
%  tand, f, flag)
%  
%  PARAMETERS
%  
%  INPUT PARAMETERS
%  
%    * z0 : desired characteristic impedance (Ohms)
%    * w : width of coplanar line (meters)
%    * h : substrate thickness (meters)
%    * l : lentgh of coplanar line (meters)
%    * tmet : metal thickness (meters)
%    * rho : resistivity of the metalization (relative to copper)
%    * rough : conductor surface roughness (meters-rms)
%    * er : substrate relative permitivitty
%    * tand : substrate loss tangent
%    * f : synthesis frequency (Hz)
%    * flag : indicates which parameters should be optimized
%        + flag=1 : synthesize the metal width
%        + flag=1 : synthesize the substrate thickness
%        + flag=2 : synthesize the substrate dielectric constant
%        + flag=3 : synthesize the metal length
%  
%  OUTPUT VALUES
%  
%    * w_out : Calculated width of coplanar line (meters)
%    * h_out : Calculated height of coplanar line (meters)
%    * l_out : Calculated length of coplanar line (meters)
%    * er_out : Calculated substrate relative permitivitty of coplanar line
%  
%  DESCRIPTION
%  
%  Function to synthesize the physical dimenstions of a single coplanar
%  transmission line to achieve a desired characteristic impedance and electrical
%  length. The desired characteristic impedance and some of the physical
%  dimensions of the coplanar line are given as inputs and the remaining
%  dimensions are calculated.
%  
%  
%                   |<--W-->|
%                    _______
%                   | metal | <- rho,rough
%      ----------------------------------------------
%     (  dielectric                         /|\     (
%      )   er,tand                       H   |       )
%     (                                     \|/     (
%      ----------------------------------------------
%      XXXXXXXXXXXXXXXXXX ground XXXXXXXXXXXXXXXXXXXX
%  
%  
%  
%  EXAMPLE
%  
%   % mils to meters conversion factor
%  sf=25.4e-6;
%   % desired impedance
%  z0=75;
%   % desired electrical length (degrees)
%  elen=90;
%   % width (we'll synthesize this)
%  w=0;
%   % height
%  h=62*sf;
%   % length
%  l=1100*sf;
%   % metal thickness
%  tmet=1.4*sf;
%   % resitivity
%  rho=1.72e-8;
%   % metal surface roughness
%  rough=0.05*sf;
%   % relative dielectric constant
%  er=4.5;
%   % loss tangent
%  tand=0.01;
%   % frequency
%  f=2.4e9;
%   % synthesize width
%  flag=0;
%  [w_out,h_out,l_out,er_out] = ...
%    coplanar_syn(z0,elen,w,h,l,tmet,rho,rough,er,tand,f,flag);
%  disp(sprintf('w=%g mils, l=%g mils',w_out/sf,l_out/sf));
%  disp(sprintf('er=%g',er_out));
%  disp(sprintf('z0=%g ohms, length=%g degrees',z0,elen));
%  
%  SEE ALSO
%  
%  air_coil_calc
%  air_coil_syn
%  coax_calc
%  bars_calc
%  coax_syn
%  coplanar_calc
%  coplanar_syn
%  coupled_microstrip_calc
%  coupled_microstrip_syn
%  coupled_stripline_calc
%  coupled_stripline_syn
%  ic_microstrip_calc
%  ic_microstrip_syn
%  microstrip_calc
%  microstrip_syn
%  parallel_rc_p2s
%  parallel_rc_s2p
%  parallel_rl_p2s
%  parallel_rl_s2p
%  stripline_calc
%  stripline_syn
%  
%  AUTHOR
%  
%  Dan McMahill
%  
%  BUGS
%  
%  None known
%  
%  Please report any bugs on the Wcalc Sourceforge Project Page
%  


% Copyright (c) 2001-2009 Dan McMahill
% All rights reserved.
%
% This program is free software; you can redistribute it and/or modify
% it under the terms of the GNU General Public License as published by
% the Free Software Foundation; version 2 of the License.
% 
% This program is distributed in the hope that it will be useful,
% but WITHOUT ANY WARRANTY; without even the implied warranty of
% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
% GNU General Public License for more details.
% 
% You should have received a copy of the GNU General Public License
% along with this program; if not, write to the Free Software
% Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
%
